pkg/repository/udmrepo/kopialib/backend/file_system.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package backend
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/kopia/kopia/repo/blob/filesystem"
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/velero/pkg/repository/udmrepo"
+)
+
+type FsBackend struct {
+	options filesystem.Options
+}
+
+const (
+	defaultFileMode = 0o600
+	defaultDirMode  = 0o700
+)
+
+func (c *FsBackend) Setup(ctx context.Context, flags map[string]string) error {
+	path, err := mustHaveString(udmrepo.StoreOptionFsPath, flags)
+	if err != nil {
+		return err
+	}
+
+	prefix := optionalHaveString(udmrepo.StoreOptionPrefix, flags)
+
+	c.options.Path = filepath.Join(path, prefix)
```

Comment:
Could you elaborate on how the prefix is used?  Looks like it's a sub-directory for kopia, not really a prefix.

Do we really need it?

Reply:
The prefix for file system is for the same purpose as other storages. It is not Kopia specific.

For example, for Restic + S3, the prefix is like `Restic/<BSL.Spec.VolumeNamespace>`, that is, we will find the backup data under `<bucket name>/Restic/<BSL.Spec.VolumeNamespace>`.
For unified repo, we use the similar prefix pattern, the data can be found under `<bucket name>/unified-repo/<BSL.Spec.VolumeNamespace>`. 

The prefix pattern works for all kinds of storages, include file system. Therefore, with this prefix, for a file system, the data is saved under `<repo root folder>/unified-repo/<BSL.Spec.VolumeNamespace>`

Reply:
make sense, thanks.